File: modules/wem-api/src/main/java/com/enonic/wem/api/schema/content/form/Layout.java
package com.enonic.wem.api.schema.content.form;


public abstract class Layout
    extends FormItem
{
    Layout( final String name )
    {
        super( name );
    }


    @Override
    FormItemPath resolvePath()
    {
        return resolveParentPath();
    }

    public abstract FormItem getFormItem( final String name );
}

File: modules/admin/admin-ui/src/main/resources/web/admin/common/js/form/Layout.ts
module api.form {

    export class Layout extends FormItem {

        constructor(name: string) {
            super(name);
        }

        public toLayoutJson(): api.form.json.FormItemTypeWrapperJson {

            if (api.ObjectHelper.iFrameSafeInstanceOf(this, FieldSet)) {
                return (<FieldSet>this).toFieldSetJson();
            }
            else {
                throw new Error("Unsupported Layout: " + this);
            }
        }

        equals(o: api.Equitable): boolean {

            if (!(api.ObjectHelper.iFrameSafeInstanceOf(o, Layout))) {
                return false;
            }

            return super.equals(o);
        }
    }
}